driver/controller.go

```diff
@@ -206,6 +208,90 @@ func (s controllerService) validateContentSource(ctx context.Context, req *csi.C
 	return nil, "", status.Errorf(codes.InvalidArgument, "not a proper volume source %v", volumeSource)
 }
 
+// CreateSnapshot creates a logical volume snapshot.
+func (s controllerService) CreateSnapshot(ctx context.Context, req *csi.CreateSnapshotRequest) (*csi.CreateSnapshotResponse, error) {
+	// Since the kubernetes snapshots are Read-Only, we set accessType as 'ro' to activate thin-snapshots as read-only volumes
+	accessType := "ro"
+	// Set snaptype as 'thin' to activate thin-snapshots.
+	// TODO: When adding support for thick-snapshots, set this option as configurable.
+	snapType := "thin"
+
+	ctrlLogger.Info("CreateSnapshot called",
+		"name", req.GetName(),
+		"source_volume_id", req.GetSourceVolumeId(),
+		"parameters", req.GetParameters(),
+		"num_secrets", len(req.GetSecrets()))
+
+	if req.GetSourceVolumeId() == "" {
+		return nil, status.Error(codes.InvalidArgument, "missing source volume id")
+	}
+
+	if req.GetName() == "" {
+		return nil, status.Error(codes.InvalidArgument, "missing name")
+	}
+
+	name := strings.ToLower(req.GetName())
+	sourceVolID := req.GetSourceVolumeId()
+	sourceVol, err := s.lvService.GetVolume(ctx, sourceVolID)
+	if err != nil {
+		if errors.Is(err, k8s.ErrVolumeNotFound) {
+			return nil, status.Error(codes.NotFound, "failed to find source volumes")
+		}
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+	if err != nil {
+		return nil, status.Error(codes.NotFound, "failed to find source volumes")
+	}
```

Comment:
- handled already

Reply:
Nice catch, will get rid of the double check :+1: